aws-cognito-idp/src/main/java/com/workshop/awscognitoidp/controllers/practice/SubjectController.java

```diff
@@ -0,0 +1,66 @@
+package com.workshop.awscognitoidp.controllers.practice;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.workshop.awscognitoidp.helpers.GradesEnum;
+import com.workshop.awscognitoidp.models.practice.Subject;
+import com.workshop.awscognitoidp.models.practice.SubjectStudent;
+import com.workshop.awscognitoidp.services.practice.SubjectService;
+
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequestMapping(path = "/api/v1/subjects")
+@RequiredArgsConstructor
+public class SubjectController {
+  
+  private final SubjectService subjectService;
+
+  @PostMapping
+  public ResponseEntity saveSubject(@RequestBody Subject subject){
+    return new ResponseEntity<>(
+      subjectService.saveSubject(subject),
+      HttpStatus.CREATED
+    );
+  }
+
+  @GetMapping
+  public ResponseEntity getAll(){
+    return new ResponseEntity<>(
+      subjectService.findAll(),
+      HttpStatus.OK
+    );
+  }
+
+  @PutMapping("{subjectId}/grade/{grade}/student/{studentId}")
+  public ResponseEntity saveSubjectStudent(
```

Comment:
How you are restricting access? I mean only the Trainers will have access to submit the grades of students

Reply:
@dieguini ?